cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -0,0 +1,482 @@
+  template <typename T>
+  __device__ T& element(size_type element_index) noexcept {
+    return data<T>()[element_index];
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns `string_view` to the string element at the specified index.
+   *
+   * This function accounts for the offset.
+   *
+   * @param element_index Position of the desired string
+   *---------------------------------------------------------------------------**/
+  /*
+    template <>
+    __device__ string_view& element<string_view>(
+        size_type element_index) noexcept {
+      // Fill this in
+    }
+    */
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns raw pointer to the underlying bitmask allocation.
+   *
+   * @note This function does *not* account for the `offset()`.
+   *
+   * @note If `null_count() == 0`, this may return `nullptr`.
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ bitmask_type* null_mask() const noexcept {
+    return const_cast<bitmask_type*>(
+        detail::column_device_view_base::null_mask());
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the specified child
+   *
+   * @param child_index The index of the desired child
+   * @return column_view The requested child `column_view`
+   *---------------------------------------------------------------------------**/
+  __device__ mutable_column_device_view& child(size_type child_index) const
+      noexcept {
+    return mutable_children[child_index];
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Updates the null mask to indicate that the specified element is
+   * valid
+   *
+   * @note It is undefined behavior to call this function if `nullable() ==
+   * false`.
+   *
+   * @param element_index The index of the element to update
+   *---------------------------------------------------------------------------**/
+  __device__ void set_valid(size_type element_index) const noexcept {
+    return set_bit(null_mask(), element_index);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Updates the null mask to indicate that the specified element is null
+   *
+   * @note It is undefined behavior to call this function if `nullable() ==
+   * false`.
+   *
+   * @param element_index The index of the element to update
+   *---------------------------------------------------------------------------**/
+  __device__ void set_null(size_type element_index) const noexcept {
+    return clear_bit(null_mask(), element_index);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Updates the specified bitmask element in the `null_mask()` with a
+   * new element.
+   *
```

Comment:
This needs some more explanation that it will update the whole word.

Reply:
Updated the name to `set_word`, both to make it clearer what it does, and because `element` conflicts with talking about elements of a column.